src/agent/agent_config.rs

```diff
@@ -57,10 +58,9 @@ pub type InterfaceFilterFn = Box<dyn (Fn(&str) -> bool) + Send + Sync>;
 pub struct AgentConfig {
     pub urls: Vec<Url>,
 
-    /// This is optional. Leave it as 0 for the default UDP port allocation strategy.
-    pub port_min: u16,
-    /// This is optional. Leave it as 0 for the default UDP port allocation strategy.
-    pub port_max: u16,
+    /// Controls how the UDP network stack works.
+    /// See [`UDPNetwork`]
+    pub udp_network: UDPNetwork,
```

Comment:
what's the reason to merge port_min/max into UDPNetwork?

From pion/ice, I saw they are co-existing with UdpMux.

Reply:
I think this better leverages Rust's type system to guide the user. In particular, with the UDP network setup there are two cases each of which have different configuration. When we are using a muxed connection `port_max` and `port_min` are not relevant at all. 

With `UDPNetwork` what I've done is grouped all the required configuration with each case and made them mutually exclusive.

I assume it's written the way it is in `pion/ice` because Go doesn't support sum types in the same way as Rust

Reply:
The comment in `src/lib.rs` explains this well.